elements/rh-footer/RhFooter.ts

```diff
@@ -0,0 +1,290 @@
+import type { ColorPalette } from '@patternfly/pfe-core';
+
+import { LitElement, html } from 'lit';
+import { html as staticHtml, unsafeStatic } from 'lit/static-html.js';
+import { property } from 'lit/decorators.js';
+
+import { colorContextProvider } from '@patternfly/pfe-core/decorators.js';
+import { pfelement } from '@patternfly/pfe-core/decorators.js';
+import { getRandomId } from '@patternfly/pfe-core/functions/random.js';
+import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
+
+import style from './rh-footer.css';
+import { responsiveStyles } from './rh-footer-responsive.css.js';
+
+import { tabletLandscapeBreakpoint } from '../../lib/tokens.js';
+// TODO: upstream to PFE
+import { MatchMediaController } from '../../lib/MatchMediaController.js';
+
+import './rh-footer-social-link.js';
+import './rh-footer-links.js';
+import './rh-footer-block.js';
+import './rh-footer-copyright.js';
+
+import '@patternfly/pfe-icon';
+import '@patternfly/pfe-accordion';
+
+function isHeader(tagName: string) {
+  return !!tagName.match(/^H[1-6]$/i);
+}
+
+/**
+ * @element 'rh-footer'
+ * @slot    base
```

Comment:
@heyMP going through the motions of documenting these got me to wondering how many of these we want to expose. Do we have a concrete use case for the `base` slot, which overrides all the shadow content? If not, we should consider removing it (and other such slots), considering that it's easy to add to API surface, but difficult to subtract from it. WDYT?

Reply:
@bennypowers My instinct is to get rid of extra slots, creates the potential for customizations that we may not be aware of if or when they happen.

Reply:
Let's move this into a discussion thread for since this will have an effect on how we build all of our components.

Reply:
agreed. we're within our rights to make these kinds of breaking changes during pre-release season. however, we should be up front with our users about it: "just so you know, these escape hatches are here in this version, but we've discussed removing them before 1.0, so if you rely on them, it's on you to let us know"